io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/writer/internal/StreamGmlWriter.java

```diff
@@ -1765,9 +1800,16 @@ private void writeElement(Object value, PropertyDefinition propDef, IOReporter r
 				// write no elements if there is a value or only a nil reason
 				boolean writeElements = !hasValue && !hasOnlyNilReason;
 				boolean isNil = !writeElements && (!hasValue || value == null);
+				boolean isCodespace = isIdentifier || value == null; // Not
```

Comment:
If it is about determining if a code space should be added, I guess it only makes sense to add the codespace if there is an actual value for the identifier. Otherwise you end up with an identifier that has a `codeSpace` but no value, which I don't think is the expected result.